otx/algorithms/anomaly/tasks/inference.py

```diff
@@ -192,13 +194,17 @@ def infer(self, dataset: DatasetEntity, inference_parameters: InferenceParameter
         inference = AnomalyInferenceCallback(dataset, self.labels, self.task_type)
         normalize = MinMaxNormalizationCallback()
         metrics_configuration = MetricsConfigurationCallback(
-            adaptive_threshold=config.metrics.threshold.adaptive,
-            default_image_threshold=config.metrics.threshold.image_default,
-            default_pixel_threshold=config.metrics.threshold.pixel_default,
-            image_metric_names=config.metrics.image,
-            pixel_metric_names=config.metrics.pixel,
+            task=config.dataset.task,
+            image_metrics=config.metrics.image,
+            pixel_metrics=config.metrics.get("pixel"),
         )
-        callbacks = [progress, normalize, inference, metrics_configuration]
+        post_processing_configuration = PostProcessingConfigurationCallback(
+            normalization_method=NormalizationMethod.MIN_MAX,
+            threshold_method=ThresholdMethod.ADAPTIVE,
+            manual_image_threshold=config.metrics.threshold.manual_image,
+            manual_pixel_threshold=config.metrics.threshold.manual_pixel,
+        )
+        callbacks = [progress, normalize, inference, metrics_configuration, post_processing_configuration]
```

Comment:
The order doesn't matter here. The only thing that matters is that normalization is applied before the inference callback returns the predictions, but since the normalization callback is active at `on_predict_batch_end`, and the inference callback at `on_predict_epoch_end`, this does not depend on the execution order of the callbacks.

The configuration callbacks are only active during setup, and the order in which metrics and post-processing is set up does not matter.

Reply:
thanks. Similar to anomalib, a custom loop will be a quite nice solution here